src/btree/bt_read.c

```diff
@@ -321,6 +323,38 @@ __las_page_instantiate(WT_SESSION_IMPL *session, WT_REF *ref)
             WT_ERR(
               cursor->get_value(cursor, &durable_timestamp, &prepare_state, &upd_type, &las_value));
 
+            /*
+             * If our update is a modify then squash it into a standard update. It's a problem if we
+             * need to read backwards into lookaside just to make sense of what we have in our
+             * update list.
+             */
+            if (upd_type == WT_UPDATE_MODIFY) {
+                while (upd_type == WT_UPDATE_MODIFY) {
+                    WT_ERR(__wt_update_alloc(session, &las_value, &upd, &incr, upd_type));
```

Comment:
I don't believe so. We only add it to `total_incr` after we've done the final `__wt_update_alloc` which is the size the full update occupies in cache.

Reply:
I got fooled because of the name `incr`, but you're right, `__wt_update_alloc()` doesn't increment, it assigns the final size, so the `__wt_update_alloc()` calls in the loop are overwriting `incr` every time, and so the final call is all that matters.

Could we please fix this loop (and the one in `__wt_find_lookaside_upd()`), to use something other than `incr`? Maybe split it so we're using `size` and `notused`, even, so it's clear what's going on?

I don't see any other examples, could you also check the uses of `__wt_update_alloc()` in case I missed something?

Reply:
8121676